pydoctor/model.py

```diff
@@ -808,15 +822,23 @@ def ensurePackage(self, package_full_name: str) -> _PackageT:
 
     def _introspectThing(self, thing: object, parent: Documentable, parentMod: _ModuleT) -> None:
         for k, v in thing.__dict__.items():
-            if (isinstance(v, (types.BuiltinFunctionType, types.FunctionType))
+            # TODO(ntamas): MethodDescriptorType and ClassMethodDescriptorType are Python 3.7 only.
+            if (isinstance(v, func_types)
                     # In PyPy 7.3.1, functions from extensions are not
-                    # instances of the above abstract types.
-                    or v.__class__.__name__ == 'builtin_function_or_method'):
+                    # instances of the abstract types in func_types
+                    or (hasattr(v, "__class__") and v.__class__.__name__ == 'builtin_function_or_method')):
                 f = self.Function(self, k, parent)
                 f.parentMod = parentMod
                 f.docstring = v.__doc__
                 f.decorators = None
-                f.signature = Signature()
+                try:
+                    f.signature = signature(v)
+                except Exception:
+                    f.text_signature = (getattr(v, "__text_signature__") or "") + " (invalid)"
```

Comment:
If the input is invalid, pydoctor should report a warning. This increases the chance that the developer notices the problem (for example by using `--warnings-as-errors` in their CI), since it's unlikely developers will be reading their own API docs output regularly (it is mostly for the benefit of the users).